nntrainer/layers/cl_layers/concat_cl.cpp

```diff
@@ -233,12 +233,82 @@ std::string concat_cl_axis1_kernel_ =
 })";
 
 namespace nntrainer {
-ConcatLayerCl::ConcatLayerCl() : Layer() {}
+ConcatLayerCl::ConcatLayerCl() : LayerImplCl() {}
 
 static constexpr size_t SINGLE_INOUT_IDX = 0;
 static constexpr size_t INPUT_IDX_1 = 0;
 static constexpr size_t INPUT_IDX_2 = 1;
 
+bool ConcatLayerCl::registerClKernels() {
+
+  // check if already registered
+  if (!layer_kernel_ptrs.empty())
+    return true;
```

Comment:
quick question. if kernels are already registered, why does it return true?
Is there a case where kernels are registered but not the layer?

Reply:
I apologize for the confusion. 
My question is: If `layer_kernel_ptrs` is already filled, doesn't that mean the corresponding layer is also registered?
Assuming the layer is registered, wouldn't returning true in this condition register the layer more than once in the cl_context.cpp?

https://github.com/nnstreamer/nntrainer/blob/6dc250f57117375d5d55a981976f59cdf4e8540c/nntrainer/cl_context.cpp#L60-L63

Reply:
Oh I see, you mean it should be false, not true. I'm gonna update this.